UNITTESTS/README.md

```diff
@@ -0,0 +1,357 @@
+|  /          \   Unit testing
+| /------------\
+|
+*-------------------> Amount of tests
+```
+Integration and system testing are performed in an environment with full Mbed OS included in the build.
+
+Unit testing takes place in a build environment where each C/C++ class or module is tested in isolation. This means test suites are built into separate test binaries and all access outside is stubbed to remove dependency of any specific hardware or software combination. This allows the testing to be done quickly using native compilers on the build machine.
+
+## Required tools
+
+* GNU toolchains installed.
+    * GCC 6 or later
+    * MinGW-W64 GCC-6.4.0 or MinGW-W64 GCC-7.3.0 (Windows)
+* CMake 3.0+ installed.
+* Python and pip installed.
+* gcovr and optionally virtualenv pip modules installed.
+
+### Installing dependencies on Debian/Ubuntu
+
+1. `sudo apt-get -y install build-essential cmake`
+2. Install python and pip:
+```
+sudo apt-get -y install python python-setuptools
+sudo easy_install pip
+```
+3. Install [gcovr](#installing-covr).
+4. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+### Installing dependencies on Mac OS
+
+1. Install [Homebrew](https://brew.sh/).
+2. Install gcc compilers and cmake with: `brew install gcc cmake`
+3. Install python and pip:
+```
+brew install python
+sudo easy_install pip
+```
+3. Install [gcovr](#installing-covr).
+4. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+### Installing dependencies on Windows
+
+1. Download and install [MinGW-W64](http://mingw-w64.org/).
+2. Download CMake binaries from https://cmake.org/download/ and run the installer.
+3. Download Python2.7 or Python3 from https://www.python.org/getit/ and run the installer.
+4. Add MinGW, CMake and Python into PATH if not yet.
+	a. Open Start Menu > search Environment variables > Environment Variables...
+	b. Edit 'Path' and add the directory paths e.g. (C:\mingw64\bin, C:\Program Files\CMake\bin, etc.).
+5. Install [gcovr](#installing-covr).
+6. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+### Installing covr
+
+Install gcovr code coverage tool globally with `pip install 'gcovr>=4.1` or using virtualenv:
+
+#### virtualenv
+
+1. Install virtualenv if not installed with `pip install virtualenv`
+2. Install gcovr with:
+
+**[Debian/Linux/Mac OS]**
+```
+virtualenv pyenv
+. pyenv/bin/activate
+pip install 'gcovr>=4.1'
+```
+**[Windows]**
+```
+virtualenv pyenv
+pyenv\\Scripts\\activate
+pip install 'gcovr>=4.1'
+```
+
+## Building and running unit tests
+
+> In case of running into problems see [troubleshooting](#troubleshooting) section.
+
+`UNITTESTS/mbed_unittest.py` contains testing scripts for Mbed OS unit testing. Mbed CLI supports unit testing through `mbed unittest` command with the same arguments.
+
+### Testing with Mbed CLI
+
+```
+mbed unittest test
+```
+
+##### Parameters to mbed unittest test
```

Comment:
I don't feel good to list these here because these are coming from mbed-cli and mbed-cli works with any version of Mbed OS. Then we don't have guarantee that these are like this forever.